3.analyze_data/nbconverted/make_volcano_predicted_probabilities.py

```diff
@@ -0,0 +1,133 @@
+# coding: utf-8
+
+# # Compare phenotypic expression among treatments with volcano plot visualizations
+# We compare phenotypic expression for each (phenotype, statistical test, model type, treatment type) combination
+
+# In[ ]:
+
+
+import pathlib
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+
+# ## Find the root of the git repo on the host system
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define the input dataframe from the parquet path
+
+# In[ ]:
+
+
+compdf = pd.read_parquet("statistical_test_comparisons/comparison_results.parquet")
+
+
+# ## Define and create output path
+
+# In[ ]:
+
+
+output_path = pathlib.Path("figures")
+
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[1]:
+
+
+# Bernoulli adjustment
+alpha = 0.05 / len(compdf)
+
+# Remove the zero values after the p-value adjustment
+compdf = compdf.loc[compdf["p_value"] > 0]
+significance_threshold = -np.log10(alpha)
+
+# Define the unique groups to plot
+unique_comparisons = set(zip(compdf["phenotype"], compdf["treatment_type"], compdf["model_type"], compdf["statistical_test"]))
+
+# Specify colors according to categories
+colors = {
+'x_below_threshold': 'green',        # Color for x < 0 and y > significance_threshold
+'x_above_threshold': 'red',       # Color for x > 0 and y > significance_threshold
+'y_below_threshold': 'blue'          # Color for y < significance_threshold
+}
+
+# Convert the p-values to negative log p-values
+compdf["neg_log_p_value"] = -np.log10(compdf["p_value"])
+
+# Specify the (x,y) thresholds for coloring the plots
+compdf["category"] = compdf.apply(lambda row:
+    'x_below_threshold' if row['comparison_metric_value'] < 0 and row['neg_log_p_value'] > significance_threshold else
+    'x_above_threshold' if row['comparison_metric_value'] > 0 and row['neg_log_p_value'] > significance_threshold else
+    'y_below_threshold',
+    axis=1
+)
+
+for comp in unique_comparisons:
+
+    fig, ax = plt.subplots(figsize=(15, 10))
+    groupdf = compdf.copy()
+
+    # Seperate the croups we are intersted in plotting
+    groupdf = groupdf.loc[((groupdf["phenotype"] == comp[0]) & (groupdf["treatment_type"] == comp[1]) & (groupdf["model_type"] == comp[2]) & (groupdf["statistical_test"] == comp[3]))]
+
+    # Create the scatter plot with size and color differentiation
+    sns.scatterplot(
+        data=groupdf,
+        x="comparison_metric_value",
+        y="neg_log_p_value",
+        hue="category",
+        palette=colors,
+        legend=False,
+        ax = ax
+    )
+
+    # Add threshold lines
+    plt.axhline(
```

Comment:
It might also be good to include the vertical threshold lines as well like traditional volcano plots (see example):

![image](https://github.com/WayScience/JUMP-single-cell/assets/107513215/02e7a430-2c1c-4f6d-b7ee-c25887fc1029)

It would also help to really see what compounds are most significantly different from DMSO. 

If it isn't possible in this application, no worries, but it would really drive home the point of significant compounds that we should look at.